### DISCUSSION TITLE
If an update fails?

### DISCUSSION THREAD

**Discussion by dave42w** [SeaQL/sea-orm#1351] [Q&A] [answered]
From the example code
>>>>
let pear: Option<fruit::Model> = Fruit::find_by_id(28).one(db).await?;

// Into ActiveModel
let mut pear: fruit::ActiveModel = pear.unwrap().into();

// Update name attribute
pear.name = Set("Sweet pear".to_owned());

// Update corresponding row in database using primary key value
let pear: fruit::Model = pear.update(db).await?;
<<<<
How do I check if the pear.update(db) failed? 

eg the example code is not protected by a transaction so if another thread deleted the pear between the find and the update then I assume the update would fail, but I'm not sure how to check for this.

Basically, I was looking for an equivalent to the DeleteResult where I can check rows_affected.

Thanks.

**Comment by dave42w** [comment, marked as answer]
Sorry. Just realised I'm missing the point of await returning a Result (I think).

**Reply by billy1624** [reply]
No worries! Thanks for the updates :)
